bridge/discord/discord.go

```diff
@@ -94,29 +94,48 @@ func (b *Bdiscord) Connect() error {
 	serverName := strings.Replace(b.GetString("Server"), "ID:", "", -1)
 	b.nick = userinfo.Username
 	b.userID = userinfo.ID
+
+	// Try and find this account's guild, and populate channels
 	b.channelsMutex.Lock()
 	for _, guild := range guilds {
-		if guild.Name == serverName || guild.ID == serverName {
-			b.channels, err = b.c.GuildChannels(guild.ID)
-			if err != nil {
-				break
-			}
-			b.guildID = guild.ID
-			guildFound = true
+		// Skip, if the server name does not match the visible name or the ID
+		if guild.Name != serverName && guild.ID != serverName {
+			continue
+		}
+
+		// Complain about an ambiguous Server setting. Two Discord servers could have the same title!
+		// For IDs, practically this will never happen. It would only trigger if some server's name is also an ID.
+		if b.guildID != "" {
+			return errors.Errorf("There are multiple Discord servers with the same name %#v, expected to see only one", serverName)
```

Comment:
```suggestion
			return fmt.Errorf("There are multiple Discord servers with the same name %s, expected to see only one", serverName)
```

Reply:
this way we don't need to pull in another package

Reply:
`%#v` will format strings with proper quoting and escaping, so it looks like `with the same name "Server Name Here", expected to see only one`, instead of `with the same name Server Name Here, expected to see only one`. Are you sure you want `%s`?

Reply:
No good point for %#v 👍